internal/webhook/external/providers/github/provider.go

```diff
@@ -0,0 +1,102 @@
+package github
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"os"
+
+	gh "github.com/google/go-github/v71/github"
+
+	xhttp "github.com/akuity/kargo/internal/http"
+)
+
+var (
+	// ErrMissingSignature is returned when the 'X-Hub-Signature-256'
+	// header is not found or empty.
+	ErrMissingSignature = errors.New("missing signature")
+	// ErrInvalidSignature is returned when the 'X-Hub-Signature-256'
+	// header is not the value that was expected.
+	ErrInvalidSignature = errors.New("invalid signature")
+	// ErrSecretUnset is returned when the 'GH_WEBHOOK_SECRET'
+	// environment variable is empty.
+	ErrSecretUnset = errors.New("secret is unset")
+	// ErrUnsupportedEventType is returned when the received
+	// request body does not conform to a github push event.
+	ErrUnsupportedEventType = errors.New("unsupported event type")
+)
```

Comment:
I would probably not bother assigning these errors to package level vars. It seems they're mostly used in one place each and always in spots where there wouldn't be any real disadvantage to just saying `errors.New("reason")` inline.